src/connection/mod.rs

```diff
@@ -13,7 +13,7 @@ use tokio::sync::mpsc;
 
 #[derive(Debug)]
 pub enum ConnectionIncomingMessage<T: Transport, L: LoginCredentials> {
-    IncomingMessage(ServerMessage),
+    IncomingMessage(Box<ServerMessage>),
```

Comment:
I think it was actually Clippy that suggested this in the first place. If I remember correctly, it was a case of „you got one variant that‘s huge, and the other ones are tiny, if you make this a `Box`, all the variants will be small so the size taken by an instance of the enum can be small.“